### PR TITLE
parallel download support [SITL-248]

### PR DESCRIPTION
**Pull request by silverjam**
awscli:

```
$ time aws s3 cp s3://esthri-test/256MiB.bin 256MiB.bin.3
download: s3://esthri-test/256MiB.bin to ./256MiB.bin.3

real    0m6.541s
user    0m2.832s
sys     0m1.477s
```

Esthri with this PR:
```
$ time cargo run --release -- get 256MiB.bin.2 --bucket esthri-test --key 256MiB.bin
    Finished release [optimized] target(s) in 0.12s
     Running `env RUST_TEST_THREADS=1 target/release/esthri get 256MiB.bin.2 --bucket esthri-test --key 256MiB.bin`
[2021-05-26T05:40:42Z INFO  esthri] Starting, using region: UsWest2...
[2021-05-26T05:40:42Z INFO  esthri] get: bucket=esthri-test, key=256MiB.bin, file=256MiB.bin.2

real    0m4.572s
user    0m1.729s
sys     0m1.459s
```

Old code before this PR:
```
$ time cargo run --release -- get 256MiB.bin.2 --bucket esthri-test --key 256MiB.bin
    Finished release [optimized] target(s) in 0.11s
     Running `env RUST_TEST_THREADS=1 target/release/esthri get 256MiB.bin.2 --bucket esthri-test --key 256MiB.bin`
[2021-05-26T05:43:14Z INFO  esthri] Starting, using region: UsWest2...
[2021-05-26T05:43:14Z INFO  esthri] get: bucket=esthri-test, key=256MiB.bin, file=256MiB.bin.2
[2021-05-26T05:43:14Z INFO  esthri] get: bucket=esthri-test, key=256MiB.bin

real    0m18.429s
user    0m2.496s
sys     0m3.121s
```

### CODE REVIEW

File: src/esthri/src/lib.rs [hunk 1]
```diff
@@ -278,7 +278,7 @@ where
             create_chunk_upload_stream(chunk_stream, s3.clone(), upload_id.clone(), bucket, key)
                 .await;
 
-        let worker_count = Config::global().worker_count();
+        let worker_count = Config::global().worker_count(1.0);
```
Comment (jayvdb) [inline src/esthri/src/lib.rs:281]: fractional workers are a thing?

Comment (silverjam) [reply src/esthri/src/lib.rs:281]: Not sure if "workers" is the best term for this, it's basically the number of futures to run in parallel.  The worker thread pool size is ultimately sized by tokio with "magic".

As for the fractional worker/future count, no not really 😅  --  I just needed a way to size counts according to a multiplier.  I was anticipating needing to reduce the count for the next phase of this, but I can forgo this for now and figure out a better abstraction in the next PR.

Comment (notoriaga) [reply src/esthri/src/lib.rs]: Another possible thing to expose would be the tokio worker thread count (only applies to the cli). right now we use 1 per core

File: src/esthri/src/types.rs [hunk 2]
```diff
@@ -120,3 +120,64 @@ pub struct S3Object {
...
+        format!(
+            "bytes={}-{}",
+            self.offset,
+            self.offset + self.read_size() as u64 - 1,
+        )
+    }
+}
+
+impl ReadSize {
+    /// Creats an object for tracking read sizes and offsets when requesting data from S3.  Goals
```
Comment (notoriaga) [inline src/esthri/src/types.rs:144]:
```suggestion
    /// Creates an object for tracking read sizes and offsets when requesting data from S3.  Goals
```

**Review by notoriaga — APPROVED** [review]

**Review by jayvdb — APPROVED** [review]

**Pull request merged**